components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AbstractApplicationAuthenticator.java

```diff
@@ -91,6 +111,23 @@ public AuthenticatorFlowStatus process(HttpServletRequest request,
                     publishAuthenticationStepAttempt(request, context, context.getSubject(), true);
                     return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
                 } catch (AuthenticationFailedException e) {
+                    String retryPage = ConfigurationFacade.getInstance().getAuthenticationEndpointRetryURL();
```

Comment:
Again this also should be handled in BasicAuthenticator, because this logic is specific to BasicAuthenticator.

Reply:
Yes, but in reviews we decided to handle it from framework for other authenticators as we fire the events for all authenticator from the framework itself. 
As per this comment, then do we need to handle this in specific authenticators as it already handled in BasicAuthenticator.

Reply:
If the decision of the code review was to handle this logic here, you need to have all the conditions handled in basic authenticator in here as well. And you need handle the situation where the multioption is configured.